### PR TITLE
[RCIAM-96] Add IdP Source in CO Person's profile(canvas web page) in Organizational Identities table

### PR DESCRIPTION
**Pull request by ioigoume**
This fix enables the extraction of the AthenticationAuthority assertion attribute from the IdP's metadata. Then we store the url by inserting the value into the OrgIdentity table, and finally we depict in the CO Person's profile.

We need to add auth_idp_url column in org_identities table
`alter table cm_org_identities add column authn_authority varchar(1024);`

We need to request from shibboleth SP linked with COManage to fetch the data we need. The configuration should be updated by adding 
```
<!-- Map to extract attributes from SAML assertions. -->
// LEAVE ALL THE OTHER AttributeExtractor entries as is and add the following line
<AttributeExtractor type="Assertion" AuthenticatingAuthority="AuthenticatingAuthority"/>
```

As far as the old users are concerned, the source field will be updated automatically as soon as they use an IdP to log into COManage.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,3 +39,4 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - CO Person's email gets verified during the registration process
 - Add global scope for `Localization` variables of the default CO, COManage. This CO is only accessible by the platform administors.
 - Allow CO Person to view all Org Identities linked to his/her profile
+- Retrieve AuthenticatingAuthority and depict in CO Person's canvas/profile
```
Comment (NicolasLiampotis) [inline CHANGELOG.md]: Move to Added section (features)

File: app/Config/Schema/schema.xml [hunk 2]
```diff
@@ -131,6 +131,7 @@
     <field name="title" type="C" size="128" />
     <field name="o" type="C" size="128" />
     <field name="ou" type="C" size="128" />
+    <field name="auth_idp_url" type="C" size="128" />
```
Comment (NicolasLiampotis) [inline app/Config/Schema/schema.xml]: Rename to authn_authority

File: app/Lib/lang.php [hunk 3]
```diff
@@ -1020,6 +1020,7 @@
   'fd.affiliation.ep' => 'eduPersonAffiliation',
   'fd.affiliation.ep.map.desc' => 'Map the extended affiliation to this eduPersonAffiliation, see <a href="https://spaces.internet2.edu/display/COmanage/Extending+the+Registry+Data+Model#ExtendingtheRegistryDataModel-%7B%7BeduPersonAffiliation%7D%7DandExtendedAffiliations">eduPersonAffiliation and Extended Affiliations</a>',
   'fd.all' =>         'All',
+  'fd.auth_idp_url' => 'Authenticating IdP',
```
Comment (NicolasLiampotis) [inline app/Lib/lang.php]: authn_authority => 'Authenticating Authority'

File: app/Model/OrgIdentity.php [hunk 4]
```diff
@@ -145,6 +145,13 @@ class OrgIdentity extends AppModel {
         'allowEmpty' => true
       )
     ),
+    'auth_idp_url' => array(
```
Comment (NicolasLiampotis) [inline app/Model/OrgIdentity.php]: Same as above

File: app/Model/OrgIdentity.php [hunk 5]
```diff
@@ -542,7 +549,8 @@ public function updateFromEnv($id, $envAttrs) {
     $newOrgIdentity = null;  // What we decided to save as the updated record (or portion thereof)
     $newModels = array();    // List of associated models we decided to save
     $deleteCert = array();   // List of certificates that are going to be deleted
-    
+    $authIdPUrl = isset($_SERVER['AuthenticatingAuthority']) ? $_SERVER['AuthenticatingAuthority'] : "";
```
Comment (NicolasLiampotis) [inline app/Model/OrgIdentity.php]: Same as above

Comment (NicolasLiampotis) [inline app/Model/OrgIdentity.php]: Same as above

File: app/View/OrgIdentities/fields.inc [hunk 6]
```diff
@@ -418,6 +418,20 @@
...
             </div>
           </li>
+          <li>
+            <div class="field-name">
+              <?php print("Authenticating Authority(IdP)"); ?>
+            </div>
+            <div class="field-info">
+              <?php
+                if(!empty($org_identities[0]['OrgIdentity']['auth_idp_url'])) {
+                  print filter_var($org_identities[0]['OrgIdentity']['auth_idp_url'],FILTER_SANITIZE_SPECIAL_CHARS);
```
Comment (NicolasLiampotis) [inline app/View/OrgIdentities/fields.inc]: Same as above

File: app/Model/OrgIdentity.php (hunk 5 above)
Comment (NicolasLiampotis) [reply app/Model/OrgIdentity.php]: Change "" to null if things don't break

File: app/View/OrgIdentities/fields.inc [hunk 7]
```diff
@@ -418,6 +418,20 @@
               ?>
             </div>
           </li>
+          <li>
+            <div class="field-name">
+              <?php print("Authenticating Authority(IdP)"); ?>
+            </div>
+            <div class="field-info">
+              <?php
+                if(!empty($org_identities[0]['OrgIdentity']['auth_idp_url'])) {
```
Comment (NicolasLiampotis) [inline app/View/OrgIdentities/fields.inc]: Same as above

File: app/View/OrgIdentities/fields.inc (hunk 6 above)
Comment (NicolasLiampotis) [inline app/View/OrgIdentities/fields.inc]: Same as above

File: app/View/CoPeople/fields.inc [hunk 8]
```diff
@@ -1153,7 +1153,10 @@
                     <td>
                       <?php
                         if(!empty($link['OrgIdentity']['OrgIdentitySourceRecord']['OrgIdentitySource']['description'])) {
-                          print filter_var($link['OrgIdentity']['OrgIdentitySourceRecord']['OrgIdentitySource']['description'] ,FILTER_SANITIZE_SPECIAL_CHARS);
+                          print filter_var($link['OrgIdentity']['OrgIdentitySourceRecord']['OrgIdentitySource']['description'] ,FILTER_SANITIZE_SPECIAL_CHARS). "<br>";
+                        }
+                        if(!empty($link['OrgIdentity']['auth_idp_url'])) {
+                            print "( <span style='font-weight: bold;'>IdP :</span> " .filter_var($link['OrgIdentity']['auth_idp_url'] ,FILTER_SANITIZE_SPECIAL_CHARS). ") <br>";
```
Comment (NicolasLiampotis) [inline app/View/CoPeople/fields.inc]: Remove IdP label

File: app/View/OrgIdentities/fields.inc [hunk 9]
```diff
@@ -418,6 +418,20 @@
               ?>
             </div>
           </li>
+          <li>
+            <div class="field-name">
+              <?php print("Authenticating Authority(IdP)"); ?>
```
Comment (NicolasLiampotis) [inline app/View/OrgIdentities/fields.inc]: Add space - it's free :-)

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: app/View/OrgIdentities/fields.inc (hunk 7 above)
Comment (ioigoume) [reply app/View/OrgIdentities/fields.inc]: Done

File: app/View/OrgIdentities/fields.inc (hunk 6 above)
Comment (ioigoume) [reply app/View/OrgIdentities/fields.inc]: Done

File: app/Config/Schema/schema.xml [hunk 10]
```diff
@@ -131,6 +131,7 @@
     <field name="title" type="C" size="128" />
     <field name="o" type="C" size="128" />
     <field name="ou" type="C" size="128" />
+    <field name="authn_authority" type="C" size="128" />
```
Comment (NicolasLiampotis) [inline app/Config/Schema/schema.xml]: Check if max size conforms to max length of SAML entityID

File: app/Model/OrgIdentity.php [hunk 11]
```diff
@@ -542,7 +549,8 @@ public function updateFromEnv($id, $envAttrs) {
     $newOrgIdentity = null;  // What we decided to save as the updated record (or portion thereof)
     $newModels = array();    // List of associated models we decided to save
     $deleteCert = array();   // List of certificates that are going to be deleted
-    
+    $authIdPUrl = isset($_SERVER['AuthenticatingAuthority']) ? $_SERVER['AuthenticatingAuthority'] : null;
```
Comment (NicolasLiampotis) [inline app/Model/OrgIdentity.php]: Rename authIdPUrl to authnAuthority

File: app/Model/OrgIdentity.php [hunk 12]
```diff
@@ -593,6 +601,8 @@ public function updateFromEnv($id, $envAttrs) {
       // in order for SaveAssociated to know what to do).
       
       $newOrgIdentity['OrgIdentity'] = $curOrgIdentity['OrgIdentity'];
+      // Assign the authn_authority value
+      $newOrgIdentity['OrgIdentity']['authn_authority'] = $authIdPUrl;
```
Comment (NicolasLiampotis) [inline app/Model/OrgIdentity.php]: Rename authIdPUrl to authnAuthority

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

**Comment by ioigoume** [comment]
Review fixes completed.

**Review by NicolasLiampotis — APPROVED** [review]

**Comment by NicolasLiampotis** [comment]
LGTM!

**Pull request merged**